src/unix/fs.c

```diff
@@ -382,6 +449,116 @@ static ssize_t uv__fs_scandir(uv_fs_t* req) {
   return n;
 }
 
+static int uv__fs_opendir(uv_fs_t* req) {
+  DIR *dir_stream;
+  uv__dirent_t *dirent;
+  uv_dir_t *dir;
+  size_t len;
+
+  assert(req);
+
+  dir_stream = opendir(req->path);
+  if (dir_stream == NULL)
+    return -1;
+
+  len = uv__fs_direntry_size(dir_stream);
+  if (len == (size_t)-1)
+    return -1;
```

Comment:
Hum, does this need to be a runtime check? If so, we'd need to set errno before returning -1.

Reply:
I think this is just overly-cautious code and I'm not even sure what `errno` would be set to.  If you have an idea, I'll make it happen.